### DISCUSSION TITLE
Look at moving lib into a library.

### DISCUSSION THREAD

**Discussion by bcordis** [Joomla-Bible-Study/Proclaim#847] [Ideas]
This will make it easer to have them available across other components we may create.

**Comment by genu** [comment]
An idea that came to mind concerning this.

Perhaps, create a Joomla library containing all general biblestudy functionality. Similar to the way that Joomla CMS is separate from the Joomla Platform. Perhaps we can have **JBS Platform**, and **JBS Manager**. The JBS Platform can be organized similar to the way that the Joomla Platform is organized but focus on incorporating generic classes related to the component. The JBS manager, can then continue in being the component that is used to actually manage the biblestudy content in Joomla.

The idea is to have the **JBS Platform** be able to be independent of the **JBS Manager** so that anyone can actually use the **JBS Platform** to write other christian related components for Joomla.

**Comment by tomfuller2** [comment]
What a great idea Eugen. I think Brent had also thought of something like
this.

Shall we consider doing this as jbs 9?

On Tuesday, April 30, 2013, Eugen Istoc wrote:

> An idea that came to mind concerning this.
> 
> Perhaps, create a Joomla library containing all general biblestudy
> functionality. Similar to the way that Joomla CMS is separate from the
> Joomla Platform. Perhaps we can have _JBS Platform_, and _JBS Manager_.
> The JBS Platform can be organized similar to the way that the Joomla
> Platform is organized but focus on incorporating generic classes related to
> the component. The JBS manager, can then continue in being the component
> that is used to actually manage the biblestudy content in Joomla.
> 
> The idea is to have the _JBS Platform_ be able to be independent of the _JBS
> Manager_ so that anyone can actually use the _JBS Platform_ to write
> other christian related components for Joomla.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Joomla-Bible-Study/Joomla-Bible-Study/issues/127#issuecomment-17262055
> .

## 

Sent from my iPad

**Comment by bcordis** [comment]
Closing — the codebase now uses Joomla 4+ namespaces and DI container. A separate library package is no longer warranted.
